app/javascript/controllers/image_loader_controller.js

```diff
@@ -0,0 +1,65 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["link"]
+
+  connect() {
+    this.dots = ""
+    this.isPolling = false
+    this.originalText = null
+    this.loadingText = this.element.dataset.loadingText
+    this.maxedOutText = this.element.dataset.maxedOutText
+    this.errorText = this.element.dataset.errorText
+  }
+
+  async load(event) {
+    event.preventDefault()
+
+    if (this.isPolling) return
+
+    const url = this.linkTarget.href
+    this.originalText ||= this.linkTarget.textContent
+    this.isPolling = true
```

Comment:
Seems like this should at least be just below line 18.  Ideally there would be a way to make test/set really atomic, but I don't know how to do that in JavaScript (but I imagine there must be some way).

Reply:
I think this fires on the user clicking the link.  The only way we could get a race condition here is if the user could click ultra fast!  I'll move the line up, but I'm not gonna sweat figuring out how to make groups of statements atomic in JS.  (I might ask google anyway out of curiosity... it may be Atomics.compareExchange...)

Reply:
Yeah, it seems unlikely to be a problem.  It's just seems like at least setting it as close as possible makes sense and reduces the chance of someone later coming along and introducing something that causes enough of a delay that it could become a problem.  It's the kind of thing that would be really hard to debug later.

Reply:
Done.